components/molecule/tabs/src/index.js

```diff
@@ -0,0 +1,154 @@
+import React, {Component} from 'react'
+import PropTypes from 'prop-types'
+import cx from 'classnames'
+
+const TYPES = {
+  HORIZONTAL: 'horizontal',
+  VERTICAL: 'vertical',
+  FULLWIDTH: 'fullWidth'
+}
+
+const VARIANTS = {
+  HIGHLIGHTED: 'highlighted',
+  CLASSIC: 'classic'
+}
+
+/**
+ * Return classes for general object
+ * @param  {string} options.className
+ * @param  {string} options.size
+ * @param  {string} options.type
+ * @return {string}
+ */
+const getClassNames = function ({className, variant, type}) {
+  return cx(
+    'sui-MoleculeTabs',
+    `sui-MoleculeTabs--${variant}`,
+    `sui-MoleculeTabs--${type}`,
+    className
+  )
+}
+
+/**
+ * Return a boolean to render or not the icon
+ * @param  {object} options.icon
+ * @return {boolean}
+ */
+const shouldRenderIcon = function (icon) {
+  return icon
+}
+
+class MoleculeTabs extends Component {
+  state = {
+    activeTab: this.props.activeTab
+  }
+
+  _createHandleClick (index) {
+    return (event) => {
+      event.preventDefault()
+      if (this.props.items[index].enabled !== false) {
+        this.setState({ activeTab: index })
+        this.props.handleClick(index, this.props.items[index])
+      }
+    }
+  }
+
+  _renderTabs () {
+    const { items } = this.props
+    const { activeTab } = this.state
+
+    return items.map((item, index) => {
+      const tabLinkClassName = cx('sui-MoleculeTabs-button', {
+        'is-active': activeTab === index,
+        'is-disabled': item.enabled === false
+      })
+
+      return (
+        <li className='sui-MoleculeTabs-item' key={index}>
+          <button
+            className={tabLinkClassName}
+            onClick={this._createHandleClick(index)}
+            role='tab'
+          >
+            {
+              shouldRenderIcon(item.icon) &&
```

Comment:
Yes, the reason to do this was because i don't really now how icons are going to work and doing in that way allow me to improve the checks later, but right now it makes more sense like you say